### ISSUE TITLE
Not closing with single quote properly when '%'

### ISSUE THREAD

**Issue by kevindevos** [jeff-hykin/cpp-textmate-grammar#443]
Checklist

- [x] This problem exists even with the setting `"C_Cpp.enhancedColorization": "Disabled"`
- [ ] This bug exists for C
- [x] This bug exists for C++
- [ ] This bug exists for Objective-C
- [ ] This bug exists for Objective-C++

## The code with a problem is:
```
#include <iostream>
int a;
int b;
auto str = "abdce";
'%' + foo()
float c;
int d;
auto ch = 'a';
float e;
int f;
```

## It looks like:
** screenshot + theme name preferable **
![image](https://user-images.githubusercontent.com/10975732/73031487-2b115880-3e34-11ea-994f-b08afd3546f3.png)
Theme is Andromeda Bordered

## It should look like:
 
![image](https://user-images.githubusercontent.com/10975732/73031911-66605700-3e35-11ea-8c9d-d43124e50950.png)

**Comment by kevindevos** [comment]
The last version that does not have this issue from what i tested was 1.8.4.
From 1.8.6 forward the problem appears

**Comment by jeff-hykin** [comment]
Thanks for the info, just fixed it with `v1.14.22`. Let me know if the problem persists at all.

The issue was one of the old C %-escapes for printf was causing a problem. Looks like it was only designed for double quote strings.
